src/Data/String/Base.agda

```diff
@@ -90,6 +92,24 @@ intersperse sep = concat ∘′ (List.intersperse sep)
 unlines : List String → String
 unlines = intersperse "\n"
 
+module _ {p} {P : Pred Char p} (P? : Decidable P) where
+
+  wordsBy : String → List String
+  wordsBy = go [] ∘ toList where
```

Comment:
This can be fairly easily generalised to arbitrary types rather than just `Char` right? Might be worth putting in `Data.List.Base`?